lib/wasi/src/runtime/stdio.rs

```diff
@@ -0,0 +1,256 @@
+use std::{
+    io::{self, SeekFrom},
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+};
+
+use derivative::Derivative;
+use futures::Future;
+use wasmer_vfs::{AsyncRead, AsyncSeek, AsyncWrite};
+
+#[derive(Derivative)]
+#[derivative(Debug)]
+pub struct RuntimeStdout {
+    runtime: Arc<dyn crate::WasiRuntimeImplementation + Send + Sync + 'static>,
+    #[derivative(Debug = "ignore")]
+    writing: Option<(
+        Pin<Box<dyn Future<Output = io::Result<()>> + Send + Sync + 'static>>,
+        u64,
+    )>,
+}
+
+impl RuntimeStdout {
+    pub fn new(runtime: Arc<dyn crate::WasiRuntimeImplementation + Send + Sync + 'static>) -> Self {
+        Self {
+            runtime,
+            writing: None,
+        }
+    }
+}
```

Comment:
What is this? Don't we have the pipes already defined? Why are we not using them?

Reply:
This is due to a very awkward split where currently some STDIO functionality is placed on the `WasiRuntimeImplementation` trait.

It absolutely does not belong there, and should be refactored.

Haven't done so until now because it affects code across a few implementations (wasmer-web, Deploy, etc)

Reply:
All of this stuff was removed.